contracts/pool/src/contract.rs

```diff
@@ -603,6 +607,10 @@ impl LiquidityPoolTrait for LiquidityPool {
     ) -> SimulateReverseSwapResponse {
         let config = get_config(&env);
 
+        if ask_asset != config.token_a && ask_asset != config.token_b {
+            panic!("Trying to swap wrong asset. Aborting..")
+        }
+
```

Comment:
This is superflous. Literally 4 lines belowe we do it anyway. Modify it:
```
        let (pool_balance_offer, pool_balance_ask) = if ask_asset == config.token_b {
            (pool_balance_a, pool_balance_b)
        } else if ask_asset == config.token_a {
            (pool_balance_b, pool_balance_a)
        } else {
            panic_with_error!
        };
```
Repeat in all places.